tools/clang/lib/Parse/ParseExpr.cpp

```diff
@@ -446,24 +446,28 @@ Parser::ParseRHSOfBinaryExpression(ExprResult LHS, prec::Level MinPrec) {
         // If we're using '>>' as an operator within a template
         // argument list (in C++98), suggest the addition of
         // parentheses so that the code remains well-formed in C++0x.
-        if (!GreaterThanIsOperator && OpToken.is(tok::greatergreater))
+        if (!GreaterThanIsOperator && OpToken.is(tok::greatergreater)) {
```

Comment:
This change looks like it changes behavior, and not in a correct way. I assume the GCC warning you're fixing here is something about alignment and conditional nesting... I suspect that warning should be disabled.

Reply:
Sorry, I'm not seeing how this changes functionality? In any case I will revert as I do not want to incur undue confusion. I found this code rather difficult to read, but that could just be me.